pavelib/utils/timer.py

```diff
@@ -0,0 +1,73 @@
+"""
+Tools for timing paver tasks
+"""
+
+from datetime import datetime
+import json
+import os
+from os.path import dirname, exists
+import sys
+import wrapt
+
+PAVER_TIMER_LOG = os.environ.get('PAVER_TIMER_LOG')
+
+
+@wrapt.decorator
+def timed(wrapped, instance, args, kwargs):  # pylint: disable=unused-argument
+    """
+    Log execution time for a function to a log file.
+
+    Logging is only actually executed if the PAVER_TIMER_LOG environment variable
+    is set. That variable is expanded for the current user and current
+    environment variables. It also can have :meth:`~Datetime.strftime` format
+    identifiers which are substituted using the time when the task started.
+
+    For example, ``PAVER_TIMER_LOG='~/.paver.logs/%Y-%d-%m.log'`` will create a new
+    log file every day containing reconds for paver tasks run that day, and
+    will put those log files in the ``.paver.logs`` directory inside the users
+    home.
+
+    Must be earlier in the decorator stack than the paver task declaration.
+    """
+    start = datetime.utcnow()
+    exception_info = {}
+    try:
+        return wrapped(*args, **kwargs)
+    except Exception as exc:  # pylint: disable=broad-except
```

Comment:
This was an abundance of caution. Basically, if any of our paver tasks throws an exception for whatever reason, I wanted to capture that, because it likely would skew the timing.